flixel/input/gamepad/ButtonIndex.hx

```diff
@@ -0,0 +1,405 @@
+package flixel.input.gamepad;
+import flixel.input.gamepad.ButtonID;
+import flixel.input.gamepad.FlxGamepad.FlxGamepadAnalogStick;
+import flixel.input.gamepad.FlxGamepad.GamepadModel;
+
+/**
+ * ...
+ * @author larsiusprime
+ */
+class ButtonIndex
+{
+	public var model:GamepadModel;
```

Comment:
Actually, it makes sense to leave this settable, because users might want to change their mapping to handle various edge cases (the computer thinks it's "Generic XInput" but I KNOW it's XYZ!), and updating this directly should totally work.

IE:
```
gamepad.model = Logitech;
```

Right after you do that you've effectively remapped it to Logitech controls.